dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1460,7 +1461,39 @@ class CompileTest {
       case DeviceOperatingSystem.fuchsia:
         throw Exception('Unsupported option for Fuchsia devices');
       case DeviceOperatingSystem.macos:
-        throw Exception('Unsupported option for macOS devices');
+        unawaited(stderr.flush());
+        options.insert(0, 'macos');
+        options.add('--tree-shake-icons');
+        options.add('--split-debug-info=infos/');
+        watch.start();
+        await flutter('build', options: options);
+        watch.stop();
+        final String buildDirectoryPath = path.join(
+          cwd,
+          'build',
+          'macos',
+          'Build',
+          'Products',
+          'Release',
+        );
+        final String? appBundlePath =
+            _findIosAppInBuildDirectory(buildDirectoryPath);
+        if (appBundlePath == null) {
+          throw 'Failed to find app bundle in $buildDirectoryPath';
+        }
+        final Directory appBundle = Directory(appBundlePath);
+        final Archive archive = Archive();
+        await appBundle
+            .list(recursive: true)
+            .where((FileSystemEntity e) => e.runtimeType == File)
+            .map<ArchiveFile>((FileSystemEntity e) {
+              assert(e.runtimeType == File);
+              final File file = e as File;
+              return ArchiveFile(file.path, file.statSync().size, file.readAsBytesSync());
+            })
+            .forEach((ArchiveFile file) => archive.addFile(file));
+        releaseSizeInBytes = TarEncoder().encode(archive).length;
```

Comment:
It looks like we're producing a tar archive just to measure the bytes. Since tar archives are basically just a concatenation of the files in the archive plus a header before each one with the path/size/permission/timestamp data, as well as some zero-padding, this is roughly equivalent to just summing up the file sizes collected while iterating over the files. I'd just do that instead.

A followup thought -- unlike iOS, there are multiple ways developers can distribute a macOS app: via the app store or individually, typically in a compressed disk image.

We should probably decide how we want to measure this. One of:
1. The sum of the components of the files in the `.app` bundle. Current implementation.
2. The size of a distributable that would be uploaded to the app store (i.e. the equivalent of an IPA).
3. The size of the compressed app bundle (e.g. maybe a zip file).

Personally, option 1 or 2 seem most reasonable. The iOS code above is a bit weird; it's generating a compressed tar archive. I thought IPAs were a zipfile though, but I suppose a gzipped tar is 'close enough'. 

Maybe we do that for consistency?

Reply:
cc @jmagman

Reply:
@cbracken at the time of writing, **Option 1** is the approach that has been taken. Please comment/resolve as you see fit:
https://github.com/flutter/flutter/blob/d70df45883b810910e8b90a9e5aec1f750869b7b/dev/devicelab/lib/tasks/perf_tests.dart#L1437-L1479